src/main/java/com/minecolonies/coremod/entity/pathfinding/MinecoloniesAdvancedPathNavigate.java

```diff
@@ -810,23 +815,18 @@ protected void followThePath()
             {
                 this.path.advance();
                 wentAhead = true;
-                // Mark reached nodes for debug path drawing
-                if (AbstractPathJob.lastDebugNodesPath != null)
-                {
-                    final Node point = path.getNode(i);
-                    final BlockPos pos = new BlockPos(point.x, point.y, point.z);
-                    for (final MNode node : AbstractPathJob.lastDebugNodesPath)
-                    {
-                        if (!node.isReachedByWorker() && node.pos.equals(pos))
-                        {
-                            node.setReachedByWorker(true);
-                            break;
-                        }
-                    }
-                }
+
+                final Node point = path.getNode(i);
+                reached.add(new BlockPos(point.x, point.y, point.z));
```

Comment:
might want to avoid adding all position to a set for all entities each tick, when we don't even track any, I'd ask the path handler if there is any player tracking, aka size > 0 before storing so much